ReactCommon/jsi/jsi/jsi.cpp

```diff
@@ -140,7 +140,7 @@ Value Runtime::createValueFromJsonUtf8(const uint8_t* json, size_t length) {
   return parseJson.call(*this, String::createFromUtf8(*this, json, length));
 }
 
-Pointer& Pointer::operator=(Pointer&& other) {
+Pointer& Pointer::operator=(Pointer&& other) noexcept {
   if (ptr_) {
     ptr_->invalidate();
```

Comment:
I am concerned the code analysis warnings are incorrect, because nothing guarantees that invalidate() is noexcept.  If it does throw, then operator= or ~Pointer will std::terminate.  It seems that invalidate should be declared noexcept, although this may create migration issues until the change propagates through Hermes.

Thoughts?

Reply:
In practice, neither the Hermes nor JSC invalidate impls throw, and nothrow is a desirable property.  Making that change is a bit tricky, though, as it is a visible change to JSI, which can cause some impedance mismatches as the changes propagate across the repos.  But it is probably still worthwhile.